datasets/narrativeqa_manual/narrativeqa_manual.py

```diff
@@ -0,0 +1,342 @@
+                can be used to load the dataset via `datasets.load_dataset("narrativeqa_manual", data_dir="<path/to/folder>")."""
+
+    def _info(self):
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "kind": datasets.Value("string"),
+                    "story_url": datasets.Value("string"),
+                    "story_file_size": datasets.Value("string"),
+                    "wiki_url": datasets.Value("string"),
+                    "wiki_title": datasets.Value("string"),
+                    "story_word_count": datasets.Value("string"),
+                    "story_start": datasets.Value("string"),
+                    "story_end": datasets.Value("string"),
+                    "story_text": datasets.Value("string"),
+                    "summary": datasets.Value("string"),
+                    "summary_tokenized": datasets.Value("string"),
+                    "qaps": datasets.features.Sequence(
+                        {
+                            "question": datasets.Value("string"),
+                            "answer1": datasets.Value("string"),
+                            "answer2": datasets.Value("string"),
+                            "question_tokenized": datasets.Value("string"),
+                            "answer1_tokenized": datasets.Value("string"),
+                            "answer2_tokenized": datasets.Value("string"),
+                        }
+                    ),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLS)
+        documents = data_dir["documents"]
+        summaries = data_dir["summaries"]
+        qaps = data_dir["qaps"]
+
+        train_dict = {}
+        test_dict = {}
+        valid_dict = {}
+        with open(documents, encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(
+                csv_file, quotechar='"', delimiter=",", quoting=csv.QUOTE_ALL, skipinitialspace=True
+            )
+            _ = next(csv_reader)
+            for id_, row in enumerate(csv_reader):
+                if row:
+                    if row[1] == "train":
+                        train_dict[row[0]] = {
+                            "document_id": row[0],
+                            "kind": row[2],
+                            "story_url": row[3],
+                            "story_file_size": row[4],
+                            "wiki_url": row[5],
+                            "wiki_title": row[6],
+                            "story_word_count": row[7],
+                            "story_start": row[8],
+                            "story_end": row[9],
```

Comment:
Maybe you can define the `question-answers` field as an empty list here ?
This way you don't have to use try/except on KeyError when iterating on `qaps `